File: frontend_v2/types/product.ts
export interface Category {
  id: number;
  shopId: number;
  name: string;
  slug?: string;
  displayOrder: number;
  productCount: number;
  createdAt?: string;
  updatedAt?: string;
}

export interface CategoryRequest {
  name: string;
  displayOrder?: number;
}

export interface ProductVariant {
  id?: number;
  name: string;
  price: number;
  isAvailable?: boolean;
}

export interface ProductAddon {
  id?: number;
  name: string;
  price: number;
  isAvailable?: boolean;
}

export interface Product {
  id: number;
  shopId: number;
  name: string;
  description: string;
  price: number;
  categoryId?: number | null;
  categoryName?: string | null;
  category: string;
  imageUrl?: string;
  isEggless: boolean;
  inStock: boolean;
  availability?: boolean;
  preparationTimeHours?: number;
  weightGrams?: number;
  minAdvanceHours?: number;
  variants?: ProductVariant[];
  addons?: ProductAddon[];
}

export interface CreateProductRequest {
  name: string;
  description?: string;
  price: number;
  categoryId?: number | null;
  category?: string;
  imageUrl?: string;
  isEggless?: boolean;
  inStock?: boolean;
  availability?: boolean;
  preparationTimeHours?: number;
  weightGrams?: number;
  variants?: ProductVariant[];
  addons?: ProductAddon[];
}

File: backend/src/main/java/com/cakeplatform/api/modules/product/Product.java
package com.cakeplatform.api.modules.product;

import com.cakeplatform.api.modules.shop.Shop;
import jakarta.persistence.*;
import lombok.Data;
import com.fasterxml.jackson.annotation.JsonIgnore;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "products")
@Data
public class Product {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @JsonIgnore
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "shop_id", nullable = false)
    private Shop shop;

    @JsonIgnore
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "category_id")
    private ProductCategory category;

    @Column(nullable = false)
    private String name;

    private String description;

    @Column(columnDefinition = "TEXT")
    private String ingredients;

    @Column(columnDefinition = "TEXT")
    private String allergens;

    @Column(nullable = false)
    private BigDecimal price;

    @Column(name = "image_url")
    private String imageUrl;

    @Column(nullable = false)
    private Boolean availability = true;

    @Column(nullable = false)
    private String status = "ACTIVE";

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;

    @OneToMany(mappedBy = "product", cascade = CascadeType.ALL, orphanRemoval = true)
    private java.util.List<ProductVariant> variants = new java.util.ArrayList<>();

    @OneToMany(mappedBy = "product", cascade = CascadeType.ALL, orphanRemoval = true)
    private java.util.List<ProductAddon> addons = new java.util.ArrayList<>();

    @com.fasterxml.jackson.annotation.JsonProperty("categoryId")
    public Long getCategoryId() {
        return category != null ? category.getId() : null;
    }

    @com.fasterxml.jackson.annotation.JsonProperty("categoryName")
    public String getCategoryName() {
        return category != null ? category.getName() : null;
    }
}
